### ISSUE TITLE
feat(button-toggle): Add support for radio-button interaction pattern

### ISSUE THREAD

**Issue by jelbourn** [angular/components#20190]
`MatButtonToggle` and `MatButtonToggleGroup` previously use either a native checkbox or a radio button depending on whether the control had multiple selection. We changed this to always using a native `<button>` element with `aria-pressed` partially based on this [article by Heydon Pickering](https://inclusive-components.design/toggle-button).

I was recently given some feedback from a from a screen-reader user that, while `button[aria-pressed]` and checkbox are effectively interchangeable, using the radio-button interaction pattern is preferable for single-selection in a lot of cases. As such, we should re-add support for the radio-button pattern. I'm imagining we can do this similarly to how we support multiple interaction pattern in other components (e.g. list, chips) via selector, potentially something like
```html
<mat-button-toggle-radio-group aria-label="Text alignment">
  <mat-button-toggle-radio>Left</mat-button-toggle-radio>
  <mat-button-toggle-radio>Center</mat-button-toggle-radio>
  <mat-button-toggle-radio>Right</mat-button-toggle-radio>
</mat-button-toggle-radio-group>
```

This variant should use a native `<input type="radio">` in its template so that we get the correct keyboard interaction for free.

**Comment by jelbourn** [comment]
Looks like the MDC's button toggle will use the radio pattern for single-select toggle groups, so this will probably just be fixed by default when switching to MDC.

**Comment by amysorto** [comment]
Removing 'Fixed in MDC' label since the buttons still only rely on aria-pressed and do not use a native input

**Comment by crisbeto** [comment]
We are closing this issue, because we've determined, using a semi-automated process, that it is no longer relevant, has low engagement, or it does not have enough information to proceed. Feel free to comment if you think the issue should be reopened, or open a new issue with the latest version of our code instead.
